src/Sarif.Sarifer/SariferExtensionOptionPage.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.ComponentModel;
+using System.Runtime.InteropServices;
+
+using Microsoft.VisualStudio.Shell;
+
+namespace Microsoft.CodeAnalysis.Sarif.Sarifer
+{
+    [Guid("BB3665D5-E661-48C0-801A-19B034F3CD5F")]
+    public class SariferExtensionOptionPage : DialogPage
+    {
+        private const string CategoryName = "Sarifer Options";
+        private const string DisplayName_BackgroundAnalysisEnabled = "Enable Background Analysis";
+        private const string Description_BackgroundAnalysisEnabled = "If enable background analyzer to analyze content in editor while editing.";
```

Comment:
>If enable background analyzer to analyze content in editor while editing [](start = 70, length = 72)

If enabled, all files will be analyzed.

Reply:
This "Enable Background Analysis" is mainly about if run background analysis when user open a file in text editor and edit the text in text editor. even it's false, analyze project / analyze solution / analyze file (right click menu) still analyze files. may need a more clear description?

---
In reply to: [564950816](https://github.com/microsoft/sarif-visualstudio-extension/pull/354#discussion_r564950816) [](ancestors = 564950816)

Reply:
yes! for me, that would disable the entire sarifer extension

---
In reply to: [564956784](https://github.com/microsoft/sarif-visualstudio-extension/pull/354#discussion_r564956784) [](ancestors = 564956784,564950816)

Reply:
Looking at other options in VS, we can see that they normally use checkbox and the title/description appears one after the other.
what do u think?

---
In reply to: [564957369](https://github.com/microsoft/sarif-visualstudio-extension/pull/354#discussion_r564957369) [](ancestors = 564957369,564956784,564950816)

Reply:
As discussed offline, this is option for automatic background analysis. Updated the comments. Will look for solution for checkbox view next PR

---
In reply to: [564958085](https://github.com/microsoft/sarif-visualstudio-extension/pull/354#discussion_r564958085) [](ancestors = 564958085,564957369,564956784,564950816)

Reply:
nit: th => the

---
In reply to: [564963830](https://github.com/microsoft/sarif-visualstudio-extension/pull/354#discussion_r564963830) [](ancestors = 564963830,564958085,564957369,564956784,564950816)

Reply:
done

---
In reply to: [564965910](https://github.com/microsoft/sarif-visualstudio-extension/pull/354#discussion_r564965910) [](ancestors = 564965910,564963830,564958085,564957369,564956784,564950816)